shared/chat/conversation/info-panel/index.tsx

```diff
@@ -64,708 +108,258 @@ export type InfoPanelProps = {
-          />
-        )
-
-      case 'clear entire conversation':
-        return (
-          <CaptionedDangerIcon
-            key="clear entire conversation"
-            caption="Clear entire conversation"
-            onClick={row.onShowClearConversationDialog}
-            icon="iconfont-fire"
+        ) : (
+          <AdhocHeader
+            onShowNewTeamDialog={this.props.onShowNewTeamDialog}
+            participants={this.props.participants}
           />
-        )
+        )}
+      </Kb.Box2>
+    )
+    return header
+  }
+  _headerSection = () => {
+    return {
+      data: ['header'],
+      renderItem: this._renderHeader,
+      renderSectionHeader: () => {
+        return null
+      },
+    }
+  }
 
-      case 'hide this conversation':
-        return (
-          <CaptionedDangerIcon
-            key="hide this conversation"
-            caption="Hide this conversation"
-            onClick={row.onHideConv}
-            noDanger={true}
-            icon="iconfont-hide"
-            spinner={this.props.spinnerForHide}
-          />
-        )
+  _renderAttachmentViewSelector = () => {
+    return (
+      <AttachmentTypeSelector
+        selectedView={this.props.selectedAttachmentView}
+        onSelectView={this.props.onAttachmentViewChange}
+      />
+    )
+  }
+  _attachmentViewSelectorSection = () => {
+    return {
+      data: ['avselector'],
+      renderItem: this._renderAttachmentViewSelector,
+      renderSectionHeader: () => {
+        return null
+      },
+    }
+  }
 
-      case 'unhide this conversation':
-        return (
-          <CaptionedDangerIcon
-            key="unhide this conversation"
-            caption="Unhide this conversation"
-            onClick={row.onUnhideConv}
-            noDanger={true}
-            spinner={this.props.spinnerForHide}
-          />
-        )
+  _renderTabs = () => {
+    const tabs = this._getTabs(this._getEntityType())
+    const selected = tabs.find(tab => this._isSelected(tab.key)) || null
+    return (
+      <Kb.Box2 direction="horizontal" fullWidth={true}>
+        <Kb.Tabs tabs={tabs} selected={selected} onSelect={this._onSelectTab} style={styles.tabStyle} />
+      </Kb.Box2>
+    )
+  }
+  _tabsSection = () => {
+    return {
+      data: ['tabs'],
+      renderItem: ({item}) => null,
+      renderSectionHeader: this._renderTabs,
+    }
+  }
 
-      case 'participant count':
-        return (
-          <ParticipantCount
-            key="participant count"
-            label={row.label}
-            participantCount={row.participantCount}
-          />
-        )
+  _renderSectionHeader = ({section}) => {
+    return section.renderSectionHeader({section})
+  }
 
-      case 'small team header':
+  render() {
+    const entityType = this._getEntityType()
+    let sections = []
+    let tabsSection = this._tabsSection()
+    sections.push(this._headerSection())
+    let itemSizeEstimator
+
+    if (this._isSelected('settings')) {
```

Comment:
Can this be restructured as a switch/case statement? Then we can also add a type check to make sure we handle all cases